src/main/scala/uk/gov/nationalarchives/FileProcessor.scala

```diff
@@ -162,19 +162,20 @@ class FileProcessor(
           .flatMap { stream =>
             if (!tarEntry.isDirectory) {
               val id = uuidGenerator()
-              Stream.eval[IO, (String, FileInfo)](
-                stream.chunks
-                  .map(_.toByteBuffer)
-                  .toUnicastPublisher
-                  .use(s3.upload(uploadBucket, id.toString, tarEntry.getSize, _))
-                  .map { res =>
-                    val checksum = checksumToString(res.response().checksumSHA256())
-                    tarEntry.getName -> FileInfo(id, tarEntry.getSize, tarEntry.getName.split("/").last, checksum)
-                  }
-              )
-            } else {
-              Stream.empty
-            }
+              if (tarEntry.getSize <= 0) Stream.raiseError[IO](new Exception(s"The size of a file in this tar file is 0"))
```

Comment:
The ticket says we need to check just the docx file but this is checking every file in the tar.gz. 
It does stop us copying a zero byte file to our bucket but it also means that if we get a package with a random zero byte file in we don't care about then our lambda will fail. 

I think this might work better in the `Lambda` class.

```scala
        fileInfo <- IO.fromOption(fileNameToFileInfo.get(s"$batchRef/${payload.filename}"))(
          new RuntimeException(s"Document not found for file belonging to $batchRef")
        )
        _ <- IO.raiseWhen(fileInfo.fileSize == 0)(new Exception(s"File id ${fileInfo.id} size is zero"))
```
What do you think?

Reply:
This is a good point.  The package can contain a lot of images extracted from the source .docx but we don't care if any of these have a zero size.  We only care about the .docx and the .json, and our other json validation tests would fail if that one was zero length so it's just the payload docx that we're validating here.

Reply:
@MancunianSam how come in the Lambda as opposed to in `unarchiveAndUploadToS3`? If we do it in the Lambda, then wouldn't we have to wait until all files were downloaded and uploaded until we could throw the exception?

Reply:
That's true but we only want to throw an exception if the docx file is empty. We're not bothered if any other files are empty. And we don't know which file is the docx until we've downloaded the metadata json file which tells us which is the docx file. We don't want to just filter on `docx` or anything like that because it might be flaky. It's a bit annoying but considering this will be quite rare, I think it's worth downloading the files before we find out.